pkg/certificates/reconciler/certificates_test.go

```diff
@@ -112,53 +112,82 @@ func TestReconcile(t *testing.T) {
 	}
 
 	tests := []struct {
-		name    string
-		ctor    injection.ControllerConstructor
-		key     string
-		objects []*corev1.Secret
-		asserts map[types.NamespacedName]func(*corev1.Secret)
+		name                   string
+		key                    string
+		executeReconcilerTwice bool
+		objects                []*corev1.Secret
+		asserts                map[string]func(*testing.T, *corev1.Secret)
 	}{{
 		name:    "well formed secret CA and control plane secret exists",
 		key:     namespace + "/control-plane-ctrl",
-		ctor:    NewControllerFactory("my"),
 		objects: []*corev1.Secret{wellFormedCaSecret, wellFormedControlPlaneSecret},
-		asserts: map[types.NamespacedName]func(*corev1.Secret){
-			{
-				Namespace: namespace,
-				Name:      wellFormedCaSecret.Name,
-			}: func(secret *corev1.Secret) {
+		asserts: map[string]func(*testing.T, *corev1.Secret){
+			wellFormedCaSecret.Name: func(t *testing.T, secret *corev1.Secret) {
 				require.Equal(t, wellFormedCaSecret, secret)
 			},
-			{
-				Namespace: namespace,
-				Name:      wellFormedControlPlaneSecret.Name,
-			}: func(secret *corev1.Secret) {
+			wellFormedControlPlaneSecret.Name: func(t *testing.T, secret *corev1.Secret) {
 				require.Equal(t, wellFormedControlPlaneSecret, secret)
 			},
 		},
 	}, {
 		name:    "well formed secret CA and data plane secret exists",
 		key:     namespace + "/data-plane-ctrl",
-		ctor:    NewControllerFactory("my"),
 		objects: []*corev1.Secret{wellFormedCaSecret, wellFormedDataPlaneSecret},
-		asserts: map[types.NamespacedName]func(*corev1.Secret){
-			{
-				Namespace: namespace,
-				Name:      wellFormedCaSecret.Name,
-			}: func(secret *corev1.Secret) {
+		asserts: map[string]func(*testing.T, *corev1.Secret){
+			wellFormedCaSecret.Name: func(t *testing.T, secret *corev1.Secret) {
 				require.Equal(t, wellFormedCaSecret, secret)
 			},
-			{
-				Namespace: namespace,
-				Name:      wellFormedDataPlaneSecret.Name,
-			}: func(secret *corev1.Secret) {
+			wellFormedDataPlaneSecret.Name: func(t *testing.T, secret *corev1.Secret) {
 				require.Equal(t, wellFormedDataPlaneSecret, secret)
 			},
```

Comment:
the callbacks for the `wellFormedDataPlaneSecret`/`wellFormedCaSecret` `.Name` seem very similar